### PR TITLE
fix(metricinput): update decimal operation logic

### PR DESCRIPTION
**Pull request by saty2103**
### What does this implement/fix? Explain your changes.

...

### Does this close any currently open issues?

...

### Any other comments?

...

### Dependent PRs/Commits

...

### Describe breaking changes, if any.

...

### Checklist

Check all those that are applicable and complete.

- [ ] Merged with latest `master` branch

### CODE REVIEW

File: core/components/atoms/metricInput/MetricInput.tsx [hunk 1]
```diff
@@ -201,6 +201,11 @@ export const MetricInput = React.forwardRef<HTMLInputElement, MetricInputProps>(
 
   const onArrowClick = (e: React.MouseEvent<HTMLElement, MouseEvent>, direction: string) => {
     let newValue = Number(value || 0);
+    /**
+     * we are calculating the number of digits after decimal,
+     * to use it for rounding off the newValue post +-1
+     */
```
Comment (anuradha9712) [inline core/components/atoms/metricInput/MetricInput.tsx]: Please update this message to `To calculate number of digits after decimal`

File: core/components/atoms/metricInput/MetricInput.tsx [hunk 2]
```diff
@@ -209,7 +214,11 @@ export const MetricInput = React.forwardRef<HTMLInputElement, MetricInputProps>(
     if (disabled || readOnly || !isValid) return;
 
     newValue = direction === 'down' ? newValue - 1 : newValue + 1;
-    newValue = capMax(max, capMin(min, newValue));
+    /**
+     * we need to round off newValue, as certain calculations throw erroneous result.
+     * e.g 16.1 -1 gives 15.100000000000001 (try in chrome console)
+     */
+    newValue = capMax(max, capMin(min, +newValue.toFixed(decimalDigits)));
```
Comment (anuradha9712) [inline core/components/atoms/metricInput/MetricInput.tsx]: Please remove this comment

File: core/components/atoms/metricInput/MetricInput.tsx (hunk 1 above)
Comment (saty2103) [reply core/components/atoms/metricInput/MetricInput.tsx]: Done

File: core/components/atoms/metricInput/MetricInput.tsx (hunk 2 above)
Comment (saty2103) [reply core/components/atoms/metricInput/MetricInput.tsx]: Done

**Review by satyamyadav — APPROVED** [review]

**Pull request merged**